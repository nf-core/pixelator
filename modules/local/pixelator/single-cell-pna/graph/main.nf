process PIXELATOR_PNA_GRAPH {
    tag "${meta.id}"
    label 'process_high_memory'
    label 'process_long'

    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? 'quay.io/pixelgen-technologies/pixelator:0.21.2'
        : 'quay.io/pixelgen-technologies/pixelator:0.21.2'}"

    input:
    tuple val(meta), path(edge_list), path(panel_file), val(panel)

    output:
    tuple val(meta), path("graph/*"),              emit: all_results
    tuple val(meta), path("graph/*.pxl"),          emit: pixelfile
    tuple val(meta), path("graph/*.report.json"),  emit: report_json
    tuple val(meta), path("graph/*.meta.json"),    emit: metadata_json
    tuple val(meta), path("*pixelator-graph.log"), emit: log

    path "versions.yml",                           emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:

    prefix = task.ext.prefix ?: "${meta.id}"
    def args = task.ext.args ?: ''
    def panelOpt = (panel
        ? "--panel ${panel}"
        : panel_file ? "--panel ${panel_file}" : "")

    """
    pixelator \\
        --cores ${task.cpus} \\
        --log-file ${prefix}.pixelator-graph.log \\
        --verbose \\
        single-cell-pna \\
        graph \\
        ${panelOpt} \\
        --output . \\
        ${args} \\
        ${edge_list}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        pixelator: \$(echo \$(pixelator --version 2>/dev/null) | sed 's/pixelator, version //g' )
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"

    """
    mkdir graph
    touch graph/${prefix}.report.json
    touch graph/${prefix}.meta.json
    touch graph/${prefix}.pxl
    touch ${prefix}.pixelator-graph.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        pixelator: \$(echo \$(pixelator --version 2>/dev/null) | sed 's/pixelator, version //g' )
    END_VERSIONS
    """
}
