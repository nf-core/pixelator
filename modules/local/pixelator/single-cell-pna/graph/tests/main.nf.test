nextflow_process {

    name "Test Process PIXELATOR_PNA_GRAPH"
    script "../main.nf"
    process "PIXELATOR_PNA_GRAPH"
    tag "modules"
    tag "pixelator"
    tag "pixelator/pna"
    tag "pixelator/single_cell_pna_graph"

    test("PNA graph - small test") {
        when {
            params {
                pna_graph_component_size_min_threshold = 100
            }

            process {
                """
                input[0] = [
                    [ id:'PNA055_Sample07_filtered_S7', design:'pna-2', panel:'proxiome-immuno-155', technology:'pna' ],
                    file(params.pipelines_testdata_base_path + 'testdata/pna/modules/collapse/PNA055_Sample07_filtered_S7.collapse.parquet', checkIfExists: true),
                    [],
                    'proxiome-immuno-155',
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.metadata_json).match("metadata_json") },
                { assert snapshot(process.out.report_json).match("report_json") },
                { assert snapshot(process.out.versions).match("versions") },
                {
                    with (process.out.log) {
                        assert path(get(0).get(1)).readLines().last().contains("Finished pixelator graph")
                    }
                },
            )
        }
    }
}

