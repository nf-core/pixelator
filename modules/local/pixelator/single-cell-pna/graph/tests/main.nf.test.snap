{
    "versions": {
        "content": [
            [
                "versions.yml:md5,3233a3a62435cd2b7f758d666d88dbd1"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-26T12:00:59.148872357"
    },
    "report_json": {
        "content": [
            [
                [
                    {
                        "id": "PNA055_Sample07_filtered_S7",
                        "design": "pna-2",
                        "panel": "proxiome-immuno-155",
                        "technology": "pna"
                    },
                    "PNA055_Sample07_filtered_S7.report.json:md5,8c7cfca24198dc8d0dd2fd9551c08b97"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-24T17:10:53.237758536"
    },
    "metadata_json": {
        "content": [
            [
                [
                    {
                        "id": "PNA055_Sample07_filtered_S7",
                        "design": "pna-2",
                        "panel": "proxiome-immuno-155",
                        "technology": "pna"
                    },
                    "PNA055_Sample07_filtered_S7.meta.json:md5,a5fb5308aa82dd9b96d36defdea5e6a8"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-22T17:35:26.895238955"
    }
}